src/Interpreters/Cache/FileCacheSettings.h

```diff
@@ -32,6 +32,7 @@ struct FileCacheSettings
     size_t background_download_queue_size_limit = FILECACHE_DEFAULT_BACKGROUND_DOWNLOAD_QUEUE_SIZE_LIMIT;
 
     size_t load_metadata_threads = FILECACHE_DEFAULT_LOAD_METADATA_THREADS;
+    bool load_metadata_asynchronously = false;
```

Comment:
Before merge setting need to be enabled by default and all tests need to pass.
Then after review is approved setting needs to be disabled by default. And separate pull request created where setting is enabled by default.
So we can enable it by default in next version.

Reply:
> Before merge setting need to be enabled by default and all tests need to pass.

we can also make it randomized similar to cache policies https://github.com/ClickHouse/ClickHouse/blob/18f1ec57cb33b439bc06bad8c580f6b25631153a/docker/test/stress/run.sh#L75-L90

Reply:
Agree. Let's do it.

Reply:
I've set this to enabled by default to check and fix CI. Checking crashing gtest at the moment. Once the CI is passing and before merging, I'll add a randomization for this setting, in order to catch as many failures as possible with this setting enabled now.

Reply:
> So we can enable it by default in next version.

@kitaisreal Why do we need it?